### ISSUE TITLE
[Feature] Add the pilot id to the profile page

### ISSUE THREAD

**Issue by vicb** [skylines-project/skylines#2265]
Pilots need to enter their Skylines (numerical) id in flyxc.app if they want to appear on the map.

Currently there is no "easy" way to get the numerical id.

The only way is to navigate to your user page and get the id from the last segment of the url, i.e. `https://skylines.aero/users/4758`.

This is not super convenient for pilots that are not tech savvy - and some browsers hides the address bar by default.

Would it be possible to add a row with the id on the profile page (`https://skylines.aero/settings/profile`) ?
It's much easier to direct people there to lookup there id.

Thanks.

**Comment by Turbo87** [comment]
> Pilots need to enter their Skylines (numerical) id in flyxc.app if they want to appear on the map.

that seems like a limitation in flyxc.app to me. why does that app need the user id?

**Comment by vicb** [comment]
> that seems like a limitation in flyxc.app to me. why does that app need the user id?

How else would I retrieve the pilot's positions from flyxc.app ? i.e. the API takes pilot ids (`live/id/json`)

**Comment by vicb** [comment]
Hey @Turbo87,

I think the id would be best added to the Live Tracking section, i.e. https://skylines.aero/settings/tracking

I'd like to get a stab a this if you don't mind and would be open to merge it.

I think need to do something similar to the [tracking key](https://github.com/skylines-project/skylines/blob/master/ember/app/components/settings-panels/tracking-key.hbs). The id would be to add a panel below the tracking key with the tracking id.

A few questions:
1) Do you know if there is a Docker file that can be used to bootstrap a SkyLines server ?
2) Do you have any pointer on how to pass data to the hbs. I believe this is Ember ? A pointer to a good tutorial would help.

Thanks !

**Comment by Turbo87** [comment]
the ID is quite unrelated to the tracking though, it only seems related because apparently your app is using it for live tracking. I'm not sure if we should really make this ID such a publicly visible thing. yes, it is currently part of the URLs, but only because there is nothing else unique that we could use.

why don't you use the search API to find the right users and then add them through that?

**Comment by vicb** [comment]
> The ID is quite unrelated to the tracking though

From the user point of view it is related.

They would use their ID to display their live tracks / positions on other live tracking apps.
The wouldn't have take care about their ID if they do not want to do that.

That's why I thought it could go in the Live Tracking section.

> why don't you use the search API to find the right users and then add them through that?

That could actually be an idea.

Users are used to have something unique that identify them, like a username, a password, a key, ...

Do you think it would be possible to support searching by email in the search API ?

I think the best for users would actually to be able to search by live tracking key. 
Then apps would only have to only for the tracking key whether they want to publish or retrieve live tracks.

What do you think ?

**Comment by Turbo87** [comment]
> Do you think it would be possible to support searching by email in the search API ?

no, because emails are private information that is not shared with others.

> I think the best for users would actually to be able to search by live tracking key.

same thing, the live tracking key is private information that should not be shared with anyone as it allows write access for the account

**Comment by vicb** [comment]
> same thing, the live tracking key is private information that should not be shared with anyone as it allows write access for the account

It would be up to the user to share their tracking key.
The search API should only be able to retrieve by tracking, it would not publish the tracking key.
That would be not much different at what exists today: users already share their API key if they want to be tracked (i.e. in Skylines, FlySkyHigh, ...).

What do you think ?

**Comment by lordfolken** [comment]
Closing as **wontfix** in 2026 revitalization triage.

Showing the numeric pilot id on the profile/settings page was mainly for third-party onboarding (e.g. flyxc). Prior discussion already preferred search/API discovery over making the internal id a first-class public setting, and the id is still available from the user URL when needed.

Reopen if we later decide on a different public identifier that serves many clients cleanly.
